src/pages/api/leave-team.js

```diff
@@ -0,0 +1,49 @@
+import findUser from '~/helpers/find.user';
+import prisma from '~/prisma/client';
+
+export default async function handler(req, res) {
+  if (!req.body.id) {
+    res.status(400).send({ success: false, message: 'Invalid parameters!' });
+    return;
+  }
+
+  // Finding user and team
+  const user = await findUser(req.body.id);
+  if (!user) {
+    res.status(200).json({ success: false, message: "User doesn't exists!" });
+    return;
+  }
+
+  const team = await prisma.team.findUnique({
+    where: { id: user.teamId },
+    include: { users: true },
+  });
+  if (!team) {
+    res.status(200).json({ success: false, message: "Team doesn't exists!" });
+    return;
+  }
+
+  // Exiting if team has only 1 members
+  if (team.users.length === 1) {
+    res.status(200).json({ success: false, message: 'Cannot leave the team!' });
+    return;
+  }
+
+  // Updating user i.e. removing from team
+  await prisma.user.update({
```

Comment:
You need to put before a checking that the user.id is equal to the logged user and check again the created time!